web/streetmanager/srwr_process.php

```diff
@@ -108,37 +108,37 @@ function collateRows($row) {
 	 global $printrow;
 	 global $companies;
 
+	 $record_type = $row[1];
 	 $activityId = $row[2];
-	 if ($row[1] == '001') {
+
+	 if ($record_type == '001') { # Activity
 		$printrow[$activityId]['latest_phase'] = $row[11];
-	 } else if ($row[1] == '006') {
-		$printrow[$activityId]['project_ref'] = $row[3];
-	 } else if ($row[1] == '007') {
-	    $printrow[$activityId]['works_location_coordinates'] = $row[12];
-	    $printrow[$activityId]['location_text'] = $row[6];
-		$printrow[$activityId]['work_category'] = _workCategory($row[9]);
-		$printrow[$activityId]['work_status'] = _activityStatuses($row[10]);
-	 } else if ($row[1] == '010') {
+		$printrow[$activityId]['project_ref'] = $row[5];
+	 } else if ($record_type == '007') { # Phase
+		if ($row[7] == $printrow[$activityId]['latest_phase'] ) {
+			$printrow[$activityId]['works_location_coordinates'] = $row[12];
+			$printrow[$activityId]['location_text'] = $row[6];
+			$printrow[$activityId]['work_category'] = _workCategory($row[9]);
+			$printrow[$activityId]['work_status'] = _activityStatuses($row[10]);
+		}
+	 } else if ($record_type == '010') { # Site
 	    $printrow[$activityId]['works_location_type'] = _siteLocations($row[9]);
-	 } else if ($row[1] == '008') {
+	 } else if ($record_type == '008') { # Undertaker phase
 	   if ($row[3] == $printrow[$activityId]['latest_phase'] && $row[6] ) {
 	      $printrow[$activityId]['proposed_start_date'] = _stripTime($row[6]);
-	   } else if ($row[3] == $printrow[$activityId]['latest_phase'] && $row[13]) {
-	      $printrow[$activityId]['proposed_start_date'] = _stripTime($row[13]);
+	   } else if ($row[3] == $printrow[$activityId]['latest_phase'] && $row[4]) {
+	      $printrow[$activityId]['proposed_start_date'] = _stripTime($row[4]);
 	   }
 	   if ($row[3] == $printrow[$activityId]['latest_phase'] && $row[9]) {
 	      $printrow[$activityId]['proposed_end_date'] = _stripTime($row[9]);
-	   } else if ($row[3] == $printrow[$activityId]['latest_phase'] && $row[14]) {
-	      $printrow[$activityId]['proposed_end_date'] = _stripTime($row[14]);
+	   } else if ($row[3] == $printrow[$activityId]['latest_phase'] && $row[8]) {
+	      $printrow[$activityId]['proposed_end_date'] = _stripTime($row[8]);
 	   }
 	   if ($row[3] == $printrow[$activityId]['latest_phase'] && $row[21]) {
-		  $printrow[$activityId]['traffic_management_type'] = _trafficManagementType($row[21]);
-		  $printrow[$activityId]['close_footway'] = $row[21] == '1' ? 'Yes' : 'No';
-	   }
-	   if ($row[3] == $printrow[$activityId]['latest_phase'] && $row[29]) {
-		  $printrow[$activityId]['permit_status'] = $row[29] == '1' ? 'Accepted' : 'Unknown permit';
+		  $printrow[$activityId]['traffic_management_type'] = _trafficManagementType($row[20]);
+		  $printrow[$activityId]['close_footway'] = $row[21] ? 'No' : 'Yes';
 	   }
-	 } else if ($row[1] == '099') {
+	 } else if ($record_type == '099') { # District
 	   $companies[ $row[3] . $row[4] ] = $row[5];
```

Comment:
I think this should be 2/3/4, not 3/4/5?

Reply:
I missed a column, so I think this is right whereas my paste is wrong There is an internal ref in [2] https://srwrexport.blob.core.windows.net/aurora-srwrexport/SRWR_Data_Extract_v2.01.pdf?se=2025-06-02T15%3A05%3A09Z&sp=r&sv=2017-04-17&sr=c&sig=D384DcaPU2l93DhPfaRt%2BDIWxpj51%2BLm9TAnClXh0/M%3D